src/main/java/org/jenkinsci/plugins/workflow/flow/FlowExecution.java

```diff
@@ -248,10 +248,27 @@ public boolean blocksRestart() {
      * Looks up authentication associated with this flow execution.
      * For example, if a flow is configured to be a trusted agent of a user, that would be set here.
      * A flow run triggered by a user manually might be associated with the runtime, or it might not.
-     * @return an authentication; {@link ACL#SYSTEM} as a fallback, or {@link Jenkins#ANONYMOUS} if the flow is supposed to be limited to a specific user but that user cannot now be looked up
+     * @return an authentication; {@link ACL#SYSTEM2} as a fallback, or {@link Jenkins#ANONYMOUS2} if the flow is supposed to be limited to a specific user but that user cannot now be looked up
      */
-    public abstract @NonNull Authentication getAuthentication();
+    public /* abstract */ @NonNull Authentication getAuthentication2() {
+        return Util.ifOverridden(
+                () -> getAuthentication().toSpring(),
+                org.acegisecurity.Authentication.class,
+                getClass(),
+                "getAuthentication");
+    }
 
+    /**
+     * @deprecated use {@link #getAuthentication2()}
+     */
+    @Deprecated
+    public /* abstract */ @NonNull org.acegisecurity.Authentication getAuthentication() {
```

Comment:
This commit causes:
```
java.lang.IllegalArgumentException: The specified derived class (org.jenkinsci.plugins.workflow.cps.CpsFlowExecution) does not derive from the specified base class (org.springframework.security.core.Authentication).
	at hudson.Util.isOverridden(Util.java:1536)
	at hudson.Util.ifOverridden(Util.java:1564)
	at PluginClassLoader for workflow-api//org.jenkinsci.plugins.workflow.flow.FlowExecution.getAuthentication(FlowExecution.java:266)
	at PluginClassLoader for opentelemetry//io.jenkins.plugins.opentelemetry.job.MonitoringPipelineListener.onAtomicStep(MonitoringPipelineListener.java:209)
```

Reply:
@basil  @jglick can you please check https://issues.jenkins.io/browse/JENKINS-75532

Reply:
⇨ #391